src/Mumble.proto

```diff
@@ -20,6 +20,35 @@ message Version {
 	optional string os_version = 4;
 }
 
+// Send after the version packet, indicate the protocol preference of the client.
+// - This message is not required for text-only clients. It can be send before or after
+//   the Authenticate message. However, it has be sent prior to a client attempts to send
+//   or receive any audio data.
+// - As reply, the server will send an Capabilities message as well, with only one item in
+//   the list, indicating the protocol it determines to use.
+// - In the middle of a session, a client can re-perform this negotiation by send a valid
+//   Capabilities packet again. A server may also request the client to use another Capabilities
+//   with the protocol it'd like to use inside.
+message Capabilities {
+	enum VoiceProtocol {
+		UNDEFINED = 0;  // Placeholder used by the server to indicate it doesn't support any protocol the client has provided.
```

Comment:
If the server supports none of the offered protocols, could we simply reply with a Capabilities packet which has an empty `supported_protocols`?

Reply:
Yeah I guess that'd make sense :thinking: 
@TerryGeng thoughts?